### ISSUE TITLE
Interface jumps to the top when editing annotation below first window height

### ISSUE THREAD

**Issue by amir-zeldes** [webanno/webanno#331]
When the whole annotation area doesn't fit on one screen (necessary for coref), scrolling down and editing an annotation causes the interface to jump to the top. This makes it impossible to see the context of what you are annotating and requires the user to constantly scroll down when annotating below the first window height.

This problem is present in WebAnno V2.3.1 (where the annotation window was still used) and V3.0.0 beta 6 (right hand side annotation frame). The problem did not exist in V2.0.12.

**Comment by reckart** [comment]
We'll have to investigate what change introduced this. Maybe we had some special handling back then which restored the scrolling position after re-rendering, maybe it was another problem.

**Comment by reckart** [comment]
When I open a document longer than the screen and I make annotations, there is no jumping.

For me, the jumping starts only after pressing F5 - reloading the page.

The jumping stays if I try opening the document again from the "open" action in the action bar.

The jumping stops if I go back to the home screen and then enter into the annotation section again and open the document again.

**Comment by reckart** [comment]
Similar effect with Chrome 53.0.2785.116 (64-bit) - it seems that going back through the home page does not always fix the issue.

**Comment by reckart** [comment]
It seems that removing all the font-related stuff (per #337) fixes this issue as well... just leaving this open a little more for additional testing.

**Comment by reckart** [comment]
Seems ok now for me in Firefox, Safari and Chrome - even after reloading the page.

**Comment by reckart** [comment]
Reopening this because @amir-zeldes reports the problem still occurs to him with a 2e2537a5862d753346dc363158a9dbf34b59b69e build.

**Comment by reckart** [comment]
So I have difficulties reproducing the jumping problem with

* Windows 7 32 bit Chrome 53.0.2785.116 m
* Windows 7 32 bit Firefox 43.0.1
* Windows 7 32 bit Firefox 47.0.1
* OS X Safari 9.1.3 
* OS X Chrome 53.0.2785.116
* OS X Firefox 48.0.2

I do not see a jumping to the beginning of the page.

But do I notice a *little* bit of jumping. For some reason the layout does not seem to be 100% stable.  It appears as if the width of tokens is sometimes fluctuating a bit.  However, I would consider the unstable rendering a separate issue and not as severe as the jumping to the beginning of the page.

**Comment by reckart** [comment]
Ok, I *can* reproduce the problem again locally. I uploaded a long document and set the number of visible sentences to 100. If I scroll down to annotate a single token, the interface jumps back up.

Now if I add a *long span* that creates a vertical scroll bar for the annotation area and annotate something then, it does *not* jump up... argh!

**Comment by reckart** [comment]
Even if I delete the long span again, jumping does not come back... until I reload the page.

**Comment by amir-zeldes** [comment]
This is interesting. Sadly it's hard to test if it's a latency issue since I can't simulate the click and check the scrolling via something like curl on the server command line... I am in the same network as the server, and I can get you ping times if that's helpful. Are you saying that the jump is literally a reload of some sort, or some kind of ajax behavior?

**Comment by reckart** [comment]
I still don't know why it happens, but I might have found a way to prevent it. There is some conditional code that is used when *long span* is present. If I run some of that code unconditionally, then the jumping seems to disappear even locally.... still... this is more art than science... but let's give it a try.

**Comment by reckart** [comment]
Latest fix apparently was successful :)
